core/src/main/java/org/axonframework/commandhandling/disruptor/CommandHandlerInvoker.java

```diff
@@ -108,20 +109,23 @@ public void onEvent(CommandHandlingEntry entry, long sequence, boolean endOfBatc
      *
      * @param <T>                       The type of aggregate created by the factory
      * @param eventStore                The events store to load and publish events
+     * @param repositoryProvider        Provides repositories for given aggregate types
      * @param aggregateFactory          The factory creating aggregate instances
      * @param snapshotTriggerDefinition The trigger definition for snapshots
      * @param parameterResolverFactory  The factory used to resolve parameters on command handler methods
      * @return A Repository instance for the given aggregate
      */
     @SuppressWarnings("unchecked")
     public <T> Repository<T> createRepository(EventStore eventStore,
+                                              RepositoryProvider repositoryProvider,
```

Comment:
Note that all old method and constructor signatures need be left intact, to ensure backward compatibility. Obviously, using 'old' signatures would mean the new functionality cannot be used.